### PR TITLE
Bump up flow control defaults

### LINKED ISSUE #262: HTTP/2 download speeds / flow control settings
Which having an initial look at HTTP/3, @jlaine dug out that we seem to have signiifcantly slow uploading right now.

First thing to do would be to investigate and replicate, by eg. compare and contrast a simple upload from `httpx` vs `requests/urllib3` - does it replicate trivially and reliably? This really *shouldn't* be an issue for us so will need some looking at.

Initial thoughts on where issues could be:

* We're sending too-small chunks, and always waiting for the network to fully drain on each chunk is negatively impacting performance.
* We're sending too-small chunks, and ending up with largish computation overhead relative to data transfer as a result.
* We've got some big unseen overhead in the sync-to-async marshalling.
* Something else?

### PR DESCRIPTION
**Pull request by tomchristie**
Have confirmed that this closes #262

😊😊😊

### CODE REVIEW

File: httpx/dispatch/http2.py [hunk 1]
```diff
@@ -96,6 +96,7 @@ def init_complete(self) -> BaseEvent:
         del self.state.local_settings[h2.settings.SettingCodes.ENABLE_CONNECT_PROTOCOL]
 
         self.state.initiate_connection()
+        self.state.increment_flow_control_window(2 ** 24)
```
Comment (florimondmanca) [inline httpx/dispatch/http2.py]: Our of curiosity, what’s the default value for the window?

**Comment by florimondmanca** [comment]
Also we’re getting strange KeyErrors on CI. Not sure if this is related to #581?

File: httpx/dispatch/http2.py (hunk 1 above)
Comment (tomchristie) [reply httpx/dispatch/http2.py]: 65535

**Comment by tomchristie** [comment]
> Could you verify the results from #262 (comment) got improved?

Yup that was what I was testing against.

**Comment by tomchristie** [comment]
@florimondmanca No I don't think it's related to that. We've just got a bit of a flaky mock HTTP/2 stream implementation in our tests cases, that fails once we make this change. I've cut out some stuff that we don't appear to need now that we have a neater send/receive flow.

**Comment by florimondmanca** [comment]
I tested it out locally too, and can confirm the performance improvement!

Code:

```python
import asyncio
import contextlib
import httpx
import time
import requests

url = "https://quic.aiortc.org/50000000"


@contextlib.contextmanager
def timer(flavor: str):
    start = time.time()
    try:
        yield
    finally:
        elapsed = time.time() - start
        print(f"{flavor} {elapsed:.1f}s")


async def main() -> None:
    with timer("requests"):
        requests.get(url)

    async with httpx.Client() as h1_client, httpx.Client(http2=True) as h2_client:
        with timer("httpx HTTP/1.1"):
            r = await h1_client.get(url)
            assert r.http_version == "HTTP/1.1"

        with timer("httpx HTTP/2"):
            r = await h2_client.get(url)
            assert r.http_version == "HTTP/2"


asyncio.run(main())

```

Results:

```console
requests 3.3s
httpx HTTP/1.1 3.3s
httpx HTTP/2 4.3s
```

On `master` this gave me:

```
requests 3.0s
httpx HTTP/1.1 3.3s
httpx HTTP/2 30.5s
```

There's a lot of variance between runs, but yeah we're now on the same order of magnitude between HTTP/1.1 and HTTP/2! :tada:

File: httpx/dispatch/http2.py (hunk 1 above)
Comment (florimondmanca) [reply httpx/dispatch/http2.py]: Okay. Should this line require a comment of any sort? It's the kind of line I'd bump into and wonder "Why?".

**Comment by tomchristie** [comment]
> There's a lot of variance between runs

Indeed yes. I've sometime seen it a touch faster than HTTP/1.1, sometimes a touch slower. I think it comes out a shade slower overall, but that's not surprising, and it's clearly in the same range now.

**Comment by tomchristie** [comment]
A bunch of run throughs...

```shell
$ venv/bin/python ./example.py 
requests 5.7s
httpx HTTP/1.1 7.0s
httpx HTTP/2 6.1s

$ venv/bin/python ./example.py 
requests 6.3s
httpx HTTP/1.1 6.2s
httpx HTTP/2 6.1s

$ venv/bin/python ./example.py 
requests 6.3s
httpx HTTP/1.1 6.1s
httpx HTTP/2 9.2s

$ venv/bin/python ./example.py 
requests 5.3s
httpx HTTP/1.1 5.3s
httpx HTTP/2 5.2s

$ venv/bin/python ./example.py 
requests 6.6s
httpx HTTP/1.1 13.2s
httpx HTTP/2 6.6s

$ venv/bin/python ./example.py 
requests 11.9s
httpx HTTP/1.1 7.9s
httpx HTTP/2 7.4s
```

**Comment by lundberg** [comment]
> A bunch of run throughs...

[hyperfine](https://github.com/sharkdp/hyperfine) would be useful here 😉

**Pull request merged**